### PR TITLE
[jit] Add the rest of the `dict` API

### PR DESCRIPTION
**Pull request by driazati**
This adds the rest of the `dict.???` methods that were missing

Differential Revision: [D16023573](https://our.internmc.facebook.com/intern/diff/16023573/)

### CODE REVIEW

File: torch/csrc/jit/register_prim_ops.cpp [hunk 1]
```diff
@@ -1735,13 +1735,107 @@ int dictGetDefault(Stack& stack) {
...
+  } else {
+    auto erase_count = dict.erase(key);
+    TORCH_CHECK(
+        erase_count == 1, "Expected to erase 1 item, found ", erase_count);
+    push(stack, value->value());
+  }
+  return 0;
+}
+
+int dictPopDefault(Stack& stack) {
```
Comment (wanchaol) [inline torch/csrc/jit/register_prim_ops.cpp]: can you unify these two functions into a single function? they almost share the same code except the empty case.

File: torch/csrc/jit/register_prim_ops.cpp [hunk 2]
```diff
@@ -1735,13 +1735,107 @@ int dictGetDefault(Stack& stack) {
   return 0;
 }
 
+// If the key is in the dict, return it. Else set it to the default value and
+// return that.
+int dictSetDefault(Stack& stack) {
```
Comment (wanchaol) [inline torch/csrc/jit/register_prim_ops.cpp]: maybe also unify these two?

**Review by wanchaol — COMMENTED** [review]
looks good, only have nits. also clang-tidy seems complaining.

File: torch/csrc/jit/register_prim_ops.cpp [hunk 3]
```diff
@@ -1735,13 +1735,107 @@ int dictGetDefault(Stack& stack) {
...
 int dictContains(Stack& stack) {
   auto key = pop(stack);
   auto dict = pop(stack).toGenericDict();
   push(stack, dict.contains(key));
   return 0;
 }
 
+int dictClear(Stack& stack) {
+  auto dict = pop(stack).toGenericDict();
+  dict.clear();
```
[the remark this answers is not in the document]
Comment (driazati) [reply torch/csrc/jit/register_prim_ops.cpp]: `clear()` doesn't have any returns

[the remark this answers is not in the document]
Comment (wanchaol) [reply torch/csrc/jit/register_prim_ops.cpp]: I know it does not have any returns, but if you don't push the object IValue back to stack, can you do insert again after the `clear` call?

[the remark this answers is not in the document]
Comment (wanchaol) [reply torch/csrc/jit/register_prim_ops.cpp]: Oh I guess you can because the object is still in the value table.

**Review by wanchaol — APPROVED** [review]
:shipit: assuming CI passes

**Pull request closed without merging**